commit 554ff182ef53e31147e554e84bb9b428ca57fc5d
Author: Laurent Pinchart <laurent.pinchart@...>
Date:   Sun Jul 15 15:54:03 2012 +0200

    uvcvideo: Reset the bytesused field when recycling an erroneous buffer
    
    Buffers marked as erroneous are recycled immediately by the driver if
    the nodrop module parameter isn't set. The buffer payload size is reset
    to 0, but the buffer bytesused field isn't. This results in the buffer
    being immediately considered as complete, leading to an infinite loop in
    interrupt context.
    
    Fix the problem by resetting the bytesused field when recycling the
    buffer.
    
    Signed-off-by: Laurent Pinchart <laurent.pinchart@...>

diff --git a/drivers/media/video/uvc/uvc_queue.c 
b/drivers/media/video/uvc/uvc_queue.c
index 9288fbd..5577381 100644
--- a/drivers/media/video/uvc/uvc_queue.c
+++ b/drivers/media/video/uvc/uvc_queue.c
@@ -338,6 +338,7 @@ struct uvc_buffer *uvc_queue_next_buffer(struct uvc_video_queue *queue,
	if ((queue->flags & UVC_QUEUE_DROP_CORRUPTED) && buf->error) {
		buf->error = 0;
		buf->state = UVC_BUF_STATE_QUEUED;
+		buf->bytesused = 0;
		vb2_set_plane_payload(&buf->buf, 0, 0);
		return buf;
	}
